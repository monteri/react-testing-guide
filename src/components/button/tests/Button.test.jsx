import { vi } from 'vitest';
import { describe, test, expect } from 'vitest'
import { render, screen, fireEvent } from '@testing-library/react'
import userEvent from "@testing-library/user-event";
import Button from "..";

describe('Button', () => {
  test('button is rendered with title', async () => {
    render(<Button title="Test" />)
    expect(screen.getByText('Test')).toMatchSnapshot()
  })

  test('button is rendered with title', async () => {
    render(<Button title="Test" />)
    expect(screen.getByText('Test')).toBeInTheDocument()
  })

  test('button is clicked', async () => {
    const onClick = vi.fn();
    render(<Button title="Test" onClick={onClick} />)

    // fireEvent.click(screen.getByRole('button'));
    await userEvent.click(screen.getByRole('button'));

    expect(onClick).toHaveBeenCalledOnce();
  })
})
