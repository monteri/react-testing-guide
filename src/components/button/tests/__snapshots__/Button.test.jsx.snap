// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > button is rendered with title 1`] = `
<button
  role="button"
>
  Test
</button>
`;
